com.ca.lsp.cobol/lsp-service-cobol/src/test/java/com/ca/lsp/cobol/service/CopybookSearchWithoutConfigTest.java

```diff
@@ -0,0 +1,177 @@
+  DataBusBroker dataBusBroker = mock(DataBusBroker.class);
+  Provider settingsProvider = mock(Provider.class);
+  CopybookDependencyService dependencyService = mock(CopybookDependencyService.class);
+  Communications communications = mock(Communications.class);
+  private CopybookServiceImpl copybookService;
+  private static final String DOCUMENT_URI = "file:///C:/Users/test/Test.cbl";
+  Path workspaceFolderPath = null;
+  RequiredCopybookEvent requiredCopybookEvent;
+  /** Initialize the copybook service and dependency service but not the relevant folders */
+  @Before
+  public void init() {
+    workspaceFolderPath =
+        createFolderStructure(Paths.get(System.getProperty("java.io.tmpdir"), "WORKSPACE"));
+
+    copybookService =
+        new CopybookServiceImpl(dataBusBroker, settingsProvider, dependencyService, communications);
+    copybookService.setWorkspaceFolders(generateWorkspaceFolder());
+
+    requiredCopybookEvent =
+        RequiredCopybookEvent.builder()
+            .name(CPY_NAME)
+            .documentUri(DOCUMENT_URI)
+            .textDocumentSyncType("DID_OPEN")
+            .build();
+  }
+
+  @After
+  public void cleanupTempFolder() {
+    try {
+      Files.walk(workspaceFolderPath)
+          .sorted(Comparator.reverseOrder())
+          .map(Path::toFile)
+          .forEach(File::delete);
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * This test will verify that when the profile/dataset settings are not provided to the user the
+   * server will not crash and a syntax error copybook not found is shown to the user
+   */
+  @Test
+  public void findCopybookWithoutSettingsDefined() {
+    // simulate that user have not provided settings for profile and datasetlist but the
+    // preprocessor found a copybook.
+
+    // when asking for a setting config we'll give an empty object as defined in the API
+    when(settingsProvider.get()).thenReturn(ConfigurationSettingsStorable.builder().build());
+
+    try {
+      copybookService.observerCallback(requiredCopybookEvent);
+      verifyThatDepFileIsEngaged();
+      verify(communications, times(1)).notifyCopybookMessageInfo(NO_SETTINGS);
+    } catch (Exception e) {
+      fail();
+    }
+  }
+
+  /**
+   * This test verify that when the settings configuration contains just the profile name the server
+   * will not crash and a syntax error of copybook not found is shown to the user.
+   */
+  @Test
+  public void findCopybookWithOnlyProfileDefined() {
+    // simulate that user have not provided settings for profile and datasetlist but the
+    // preprocessor found a copybook.
+
+    when(settingsProvider.get()).thenReturn(new ConfigurationSettingsStorable("myProfile", null));
+
+    try {
+      copybookService.observerCallback(requiredCopybookEvent);
+      verifyThatDepFileIsEngaged();
+      verify(communications, times(1)).notifyCopybookMessageInfo(NO_DATASET_IN_SETTINGS);
+    } catch (Exception e) {
+      fail();
```

Comment:
log please

Reply:
The best way is to use `fail(e.getMessage);` here, or even something more descriptive